cmd/proxy/main.go

```diff
@@ -4,10 +4,12 @@ import (
 	"log"
 
 	"github.com/gomods/athens/cmd/proxy/actions"
+	"github.com/gomods/athens/pkg/module"
 )
 
 func main() {
-	app, err := actions.App()
+	mf := module.NewFilter()
```

Comment:
By not injecting the filter in the constructor I won't be able to drive the behaviour of the tests. Can you suggest me another approach? 
Shipping the filter file is sub optimal.

Reply:
@fedepaol Not sure if I misunderstood, but I just meant that you should call `module.NewFilter()` inside `cmd/proxy/actions/app.go` instead of `cmd/proxy/main.go`. Would that not work?

Reply:
That would obviously work, but by passing the filter to the constructor, I can provide a custom tailored one that behaves as I expect in the tests: https://github.com/gomods/athens/pull/456/files#diff-c11aa63db187809541291d77069b2dc9R24

If I don't do that, the filter is created empty and I would need to ship a `filter.conf` containing the different rules, that will be loaded in *all* the tests (unit and e2e), and that will be the same for all the scenarios.

Reply:
I understand now. @fedepaol do you need to spin up the whole App? Can you not just test the middleware by itself?

Reply:
I think it's the way testing with buffalo works. Looking at https://gobuffalo.io/en/docs/testing it says it needs a suite (https://github.com/gobuffalo/suite) to work.

The suite was already in place (see https://github.com/gomods/athens/blob/master/cmd/proxy/actions/actions_test.go#L14) 

I could test the single middleware function but I am afraid I would not have all the buffalo testing framework support (such as faking requests and asserting the responses)

Reply:
@fedepaol let's find out if we can just test the middleware _and_ be able to fake requests/responses. I really hate for us to set up the entire *buffalo.App  (with storage, workers, and everything) _*just*_ so we can test one middleware that has nothing to do with all of those dependencies. It defeats the purpose of unit testing :) 

I'm not a buffalo expert so CCing @arschles and @robjloranger to see if that's possible.

Reply:
Relevant to this, https://github.com/gomods/athens/issues/138 needs to be implemented and will suffer of the same problems.
So the suite and the app setup (**which was already in place before implementing my middleware test** and runned along with the other unit tests) will be for testing all the proxy routes and not _just one middleware_ .

Let's wait for the @arschles  and @robjloranger  , then I'll try to mock the buffalo context in order to make the test work.

Reply:
I think we can test just the middleware effectively. I had a quick look at the middlewares that ship with buffalo, and it looks pretty straight forward, see [buffalo/csrf_test.go](https://github.com/gobuffalo/buffalo/blob/master/middleware/csrf/csrf_test.go) for an idea. I should have time tonight if you want to pair up.

Reply:
Sorry I could not pair yesterday, had to get some sleep :-) 
Will take inspiration from the test you just linked to provide a custom middleware test.

Reply:
Done. The only quirk is https://github.com/gomods/athens/pull/456/commits/a244bac6f0b02e28d0276a82fc9f62a44f807a49#diff-bd90c74522e398d450dbe3611f594105R21 to prevent a null pointer crash when the middleware function was tracing